superset/assets/src/dashboard/actions/dashboardState.js

```diff
@@ -174,7 +173,13 @@ export function saveDashboardRequest(data, id, saveType, sucessMessage = 'This d
           dispatch(
             addSuccessToast(t(sucessMessage)),
           ),
-        ]).then(() => Promise.resolve(response)),
+        ]).then(() => {
+            if(saveType == SAVE_TYPE_OVERWRITE) {
+              dispatch(onSuccessReconcile(data.pub_sub_info))
+            }
+            Promise.resolve(response)
```

Comment:
why this check  
if(saveType == SAVE_TYPE_OVERWRITE) 
it should update in any case

Reply:
we don't want to update the state in case of saveAs that is not required.